### PR TITLE
Issue #9142: resolve Assertions.assertNotEquals

### PR DESCRIPTION
**Pull request by rnveach**
Issue #9142

Not as many as the other so this was manually done.

### CODE REVIEW

**Comment by rnveach** [comment]
Rebased for conflict.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 1]
```diff
@@ -1705,7 +1706,9 @@ public void testExcludeDirectoryNotMatch() throws Exception {
 
         final List<File> result = (List<File>) method.invoke(null, new File(getFilePath("")),
                 list);
-        assertNotEquals(0, result.size(), "Invalid result size");
+        assertWithMessage("Invalid result size")
+            .that(result.size())
+            .isNotEqualTo(0);
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java:1711]: isNotEmpty

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java [hunk 2]
```diff
@@ -203,8 +202,9 @@ protected URLConnection openConnection(URL u) {
                     .that(ex)
                     .hasCauseThat()
                     .isInstanceOf(IOException.class);
-            assertNotEquals("unable to get package file resources", ex.getMessage(),
-                    "Invalid exception message");
+            assertWithMessage("Invalid exception message")
+                    .that(ex.getMessage())
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java:206]: or 
```
that(ex)
 .hasMessagetThat()
```
?

lets decide

**Review by romani — CHANGES_REQUESTED** [review]
items

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 1 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java (hunk 2 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java]: Done anyways.

**Review by romani — APPROVED** [review]

**Pull request merged**